File: WatchWaves Watch App/Utilities/WaveFormatter.swift
import Foundation

enum WaveFormatter {
    static func heightString(_ meters: Double, useMetric: Bool = true) -> String {
        if useMetric {
            return String(format: "%.1fm", meters)
        } else {
            let feet = meters * 3.28084
            return String(format: "%.0fft", feet)
        }
    }

    static func periodString(_ seconds: Double) -> String {
        String(format: "%.0fs", seconds)
    }

    static func directionString(_ degrees: Double) -> String {
        CompassDirection.from(bearing: degrees).rawValue
    }

    static func windString(_ kmh: Double, useMetric: Bool = true) -> String {
        if useMetric {
            return String(format: "%.0f km/h", kmh)
        } else {
            let mph = kmh * 0.621371
            return String(format: "%.0f mph", mph)
        }
    }

    static func relativeTimeString(from date: Date) -> String {
        let seconds = Int(-date.timeIntervalSinceNow)
        if seconds < 60 { return "just now" }
        let minutes = seconds / 60
        if minutes < 60 { return "\(minutes) min ago" }
        let hours = minutes / 60
        if hours < 24 { return "\(hours) hr ago" }
        let days = hours / 24
        return "\(days)d ago"
    }

    static func temperatureString(_ celsius: Double, useMetric: Bool = true) -> String {
        if useMetric {
            return String(format: "%.0f°C", celsius)
        } else {
            let fahrenheit = celsius * 9.0 / 5.0 + 32
            return String(format: "%.0f°F", fahrenheit)
        }
    }

    static func inlineSummary(height: Double, direction: Double, period: Double, useMetric: Bool = true) -> String {
        "\(heightString(height, useMetric: useMetric)) \(directionString(direction)) \(periodString(period))"
    }
}

File: web/src/utils/waveFormatter.ts
import { compassDirection } from './compassDirection'

export function heightString(meters: number, useMetric = true): string {
  if (useMetric) return `${meters.toFixed(1)}m`
  return `${(meters * 3.28084).toFixed(0)}ft`
}

export function periodString(seconds: number): string {
  return `${Math.round(seconds)}s`
}

export function directionString(degrees: number): string {
  return compassDirection(degrees)
}

export function windString(kmh: number, useMetric = true): string {
  if (useMetric) return `${Math.round(kmh)} km/h`
  return `${Math.round(kmh * 0.621371)} mph`
}

export function temperatureString(celsius: number, useMetric = true): string {
  if (useMetric) return `${Math.round(celsius)}°C`
  return `${Math.round(celsius * 9 / 5 + 32)}°F`
}

export function relativeTimeString(date: Date): string {
  const diffMs = Date.now() - date.getTime()
  const diffMin = Math.floor(diffMs / 60000)
  if (diffMin < 1) return 'just now'
  if (diffMin < 60) return `${diffMin} min ago`
  const diffHr = Math.floor(diffMin / 60)
  if (diffHr < 24) return `${diffHr} hr ago`
  return `${Math.floor(diffHr / 24)}d ago`
}

export function inlineSummary(height: number, direction: number, period: number): string {
  return `${heightString(height)} ${directionString(direction)} ${periodString(period)}`
}
